packages/zwave-js/src/lib/driver/Driver.ts

```diff
@@ -1711,14 +1713,15 @@ export class Driver extends TypedEventEmitter<DriverEventCallbacks> {
 							msg.command instanceof InvalidCC
 						) {
 							// If it was, we need to notify the sender that we couldn't decode the command
-							await msg
-								.getNodeUnsafe()
-								?.commandClasses.Supervision.sendReport({
-									sessionId: supervisionSessionId,
-									moreUpdatesFollow: false,
-									status: SupervisionStatus.NoSupport,
-									secure: msg.command.secure,
-								});
+							const node = msg.getNodeUnsafe();
+							await node?.commandClasses.Supervision.sendReport({
+								sessionId: supervisionSessionId,
+								moreUpdatesFollow: false,
+								status: SupervisionStatus.NoSupport,
+								requestWakeUpOnDemand:
+									node.shouldRequestWakeUpOnDemand,
```

Comment:
IMO the `Supervision.sendReport` method itself could handle the `requestWakeUpOnDemand` flag instead. If the node's `requestWakeUp` property is true, it supports the feature and there is no other `Supervision CC Report` with the flag is queued, set the `requestWakeUpOnDemand` flag on the outgoing command.

And don't forget to reset the property after requesting wake up.

Reply:
> IMO the `Supervision.sendReport` method itself could handle the `requestWakeUpOnDemand` flag instead. If the node's `requestWakeUp` property is true, it supports the feature and there is no other `Supervision CC Report` with the flag is queued, set the `requestWakeUpOnDemand` flag on the outgoing command.
Yeah, I'll move the `shouldRequestWakeUpOnDemand` logic into the `Supervision.sendReport` function. I originally made it a parameter, because I thought there might be a scenario where we wanted to force it to `false`, but I know longer see the purpose of doing that, and I just forgot to simplify it down.
> And don't forget to reset the property after requesting wake up.
The difficulty of deciding whether or not the flag should be reset is why I opted away from the flag altogether.
(See my top level [comment](https://github.com/zwave-js/node-zwave-js/pull/3332#issuecomment-921349465) where I explain why I went away from having a manual flag.)